Ollivanders/src/net/pottercraft/ollivanders2/effect/O2Effect.java

```diff
@@ -189,4 +208,95 @@ public String getDivinationText ()
          return divinationText.get(rand);
       }
    }
+
+   /**
+    * Do any on damage effects
+    *
+    * @param event the event
+    */
+   public abstract void doOnDamage (@NotNull EntityDamageByEntityEvent event);
```

Comment:
Can we put a default empty implementation on all of these and then only override the ones we want to use instead of overriding all of them?

Reply:
We can. I did this so I wouldn't forget one but the list ended up longer than I though it would.